src/main/java/uk/ac/ic/wlgitbridge/bridge/Bridge.java

```diff
@@ -263,6 +263,21 @@ public void startBackgroundJobs() {
         gcJob.start();
     }
 
+    public boolean healthCheck() {
+       try {
+         dbStore.getNumProjects();
+         File rootDirectory = new File("/");
+         if (!rootDirectory.exists()) {
+           throw new Exception("bad filesystem state, root directory does not exist");
```

Comment:
This would indeed be bad! D: -- Is that what happened previously?

Reply:
Nah, what happened was the process got into a weird memory-bloat state that stopped it from serving any http requests. So the idea here is to do some full-system work that would at least check that the java process can talk to the database and the filesystem. 

If this isn't working, the request will probably time out, and the monitoring layer can take action.

So the specifics here are not that important really, we just want a general check :)